blackboxopt/optimizers/testing.py

```diff
@@ -148,6 +150,55 @@ def is_deterministic_with_fixed_seed(
     return True
 
 
+def is_deterministic_with_fixed_seed_and_multiple_evaluations(
+    optimizer_class: Union[
+        Type[SingleObjectiveOptimizer], Type[MultiObjectiveOptimizer]
+    ],
+    optimizer_kwargs: dict,
+) -> bool:
+    """Check if optimizer is deterministic, even after multiple evaluations.
+
+    Repeatedly initialize the optimizer with the same parameter space and a fixed seed,
+    then report a number of evaluations. The configuration of all final evaluations
+    should be equal.
+
+    By doing multiple evaluations, this tests covers effects that become visible after
+    a while, e.g. only after stages got completed in staged iteration samplers.
+
+    Args:
+        optimizer_class: Optimizer to test.
+        optimizer_kwargs: Expected to contain additional arguments for initializing
+            the optimizer. (`search_space` and `objective(s)` are set automatically
+            by the test.)
+
+    Returns:
+        `True` if the test is passed.
+    """
+    final_configurations = []
+    for _ in range(2):
+        space = ps.ParameterSpace(seed=42)
+        space.add(ps.ContinuousParameter("p1", [0, 1]))
+
+        opt = _initialize_optimizer(
+            optimizer_class,
+            optimizer_kwargs,
+            objective=Objective("loss", False),
+            objectives=[Objective("loss", False)],
+            space=space,
+            seed=42,
+        )
+
+        for _ in range(15):
+            es = opt.generate_evaluation_specification()
+            evaluation = es.create_evaluation(objectives={"loss": 0.42})
```

Comment:
Not sure if that is an issue, but having the same loss for all evaluations relies on the assumption that the sorting in BOHB is stable, which might be true or not. To fix this potential issue, simply using the iteration index as the loss, would make the sorting unique.